src/data/matrix/basic.lean

```diff
@@ -471,66 +466,224 @@ end semiring
 section homs
 
 -- TODO: there should be a way to avoid restating these for each `foo_hom`.
```

Comment:
Do you think this collection of lemmas can be removed (in a follow-up PR), now that we have `map_matrix` for each of these `foo_hom`s?

Reply:
No, but these can maybe be removed if we mark `map_zero` simp, because simp know how to solve `⇑f 0 = 0` for these homs

Reply:
`map_zero` is already `simp`, so I think we're ready for it :)

Reply:
It seems that a lot still compiles after 69a1d9ce32, so I'll make a PR for that soon.